src/posix.h

```diff
@@ -0,0 +1,57 @@
+#ifndef BENCHMARK_POSIX_H_
+#define BENCHMARK_POSIX_H_
+
+#include <stdlib.h>
+#include <stdio.h> // fileno, _fileno
+#include <string.h>  // _stricmp
+
+#include "internal_macros.h"
+
+#ifdef OS_WINDOWS
+#include <io.h>  // _isatty
+#else
+#include <strings.h>  // strcasecmp
+#include <unistd.h>   // isatty
+#endif
+
+namespace benchmark {
+namespace posix {
```

Comment:
My model here was Google Test's [internal/gtest-port.h](https://code.google.com/p/googletest/source/browse/trunk/include/gtest/internal/gtest-port.h#2239). I'll have a ponder about an alternate namespace name. Would you be open to ditching the nested namespace altogether?